generator/mysql/query_set.go

```diff
@@ -27,39 +29,77 @@ ORDER BY table_name;
 		return nil, fmt.Errorf("failed to query %s metadata result: %w", tableType, err)
 	}
 
+	tblChan := make(chan int, len(tables))
+	errChan := make(chan error, 1)
+
+	wg := sync.WaitGroup{}
+	for i := 0; i < runtime.NumCPU(); i++ {
+		wg.Add(1)
+		go func() {
+			defer wg.Done()
+			var err1 error
+			for tblIdx := range tblChan {
+				tables[tblIdx].Columns, err1 = m.GetTableColumnsMetaData(db, schemaName, tables[tblIdx].Name)
+				if err1 != nil {
+					select {
+					case errChan <- fmt.Errorf("failed to get '%s' table columns metadata: %w", tables[tblIdx].Name, err1):
+						return
+					default:
+					}
+					return
+				}
+			}
+		}()
+	}
+
```

Comment:
Using `errgroup.Group` this code can be simplified and `tblChan` and `errChan` can be removed.

Reply:
Didn't want to add another dependency. If that's not a problem then sure, errgroup is awesome!

Reply:
It's fine, because `golang.org/x/sync` is official golang repo.